### PR TITLE
Improve code readability

### PR DESCRIPTION
**Pull request by Danielwsx64**
improve swagger modules readability.

### CODE REVIEW

File: lib/swagger/formatter.ex [hunk 1]
```diff
@@ -57,71 +38,62 @@ defmodule Xcribe.Swagger.Formatter do
     }
   end
 
-  defp format_body_params(body, request) when is_map(body) do
-    body
-    |> Map.to_list()
-    |> format_params(request)
+  defp include_params(list, params, request, inn) do
```
Comment (thiamsantos) [inline lib/swagger/formatter.ex]:
```suggestion
  defp include_params(acc, params, request, inn) do
```

:nail_care:

File: lib/swagger/formatter.ex [hunk 2]
```diff
@@ -57,71 +38,62 @@ defmodule Xcribe.Swagger.Formatter do
...
-    }
+    %{get_content_type(request) => %{"schema" => format_body_schema(body)}}
   end
 
-  defp handle_json_decode({:ok, any}), do: any
-  defp handle_json_decode({:error, %{data: data}}), do: data
+  defp format_body_schema(body) do
+    body
+    |> JSON.decode()
+    |> case do
```
Comment (thiamsantos) [inline lib/swagger/formatter.ex]: WDYT of extracting this inline case in a function?

File: lib/swagger/swagger.ex [hunk 3]
```diff
@@ -2,127 +2,122 @@ defmodule Xcribe.Swagger do
...
-      requests
-      |> Enum.sort_by(& &1.status_code)
-      |> Enum.reduce(%{}, fn x, acc ->
-        Map.put(acc, x.path, Map.merge(acc[x.path] || %{}, handle_request(x, acc)))
-      end)
+  defp paths_from_requests(requests),
+    do: Enum.reduce(requests, %{}, &include_request_into_paths/2)
 
-    Map.put(swagger_map, "paths", paths)
+  defp include_request_into_paths(%{path: path} = request, paths) do
```
Comment (iagoEffting) [inline lib/swagger/swagger.ex]: instead create two function:

- `include_request_into_paths` 
- `include_request_into_given_path` 

you can use `pattern match` to make this decision for you

```elixir
defp include_request(%{path: path} = request, paths) do
  # ...
end

defp include_request(path, %{verb: verb} = request) do
  # ...
end

File: lib/swagger/swagger.ex [hunk 4]
```diff
@@ -2,127 +2,122 @@ defmodule Xcribe.Swagger do
...
-  defp handle_request(true, swagger_paths, request), do: add_response(swagger_paths, request)
-  defp handle_request(false, _swagger_paths, request), do: format_request(request)
-
-  defp add_response(swagger_map, request) do
-    original_request = swagger_map[request.path][request.verb]
+  defp build_path_item_object(%{verb: verb} = request),
+    do: %{verb => build_operation_object(request)}
 
+  defp build_operation_object(request) do
     %{
```
Comment (iagoEffting) [inline lib/swagger/swagger.ex]: you can extract this map to a function to make all function in this pipe in one layer of abstraction

```elixir
request
|> prepare_open_api_request
|> parameters_if_needed(request)
|> request_body_if_needed(request)
|> security_if_needed(request)

# ...
def prepare_open_api_request(request) do
   %{
     "summary" => Descriptor.get_action_description(request),
      "description" => Descriptor.get_request_description(request),
      "responses" => Formatter.format_responses(request)
    }
end
```

File: lib/swagger/swagger.ex [hunk 5]
```diff
@@ -2,127 +2,122 @@ defmodule Xcribe.Swagger do
...
   end
 
-  defp put_parameters_if_needed(swagger, _), do: swagger
+  defp include_parameters(false, openapi_object, _request), do: openapi_object
 
-  defp put_request_body_if_needed(swagger, %{request_body: body} = request)
-       when body not in [nil, %{}] do
-    Map.put(swagger, "requestBody", Formatter.format_body(request))
-  end
+  defp include_parameters(true, openapi_object, request),
```
Comment (iagoEffting) [inline lib/swagger/swagger.ex]: You have a break line here. You can remove the inline notation

```elixir
defp include_parameters(true, openapi_object, request) do
  Map.put(openapi_object, "parameters", Formatter.request_parameters(request))
end
```

File: lib/swagger/swagger.ex [hunk 6]
```diff
@@ -2,127 +2,122 @@ defmodule Xcribe.Swagger do
...
+    do: Map.put(openapi_object, "parameters", Formatter.request_parameters(request))
 
-  defp put_request_body_if_needed(swagger, _), do: swagger
+  defp request_body_if_needed(openapi_object, request) do
```
Comment (iagoEffting) [inline lib/swagger/swagger.ex]: you can use guard function to make this if:

```elixir
defp request_body_if_needed(openapi_object, request) 
    when request.request_body in @empty_data do
  openapi_object
end

defp request_body_if_needed(openapi_object, request)  do
  Map.put(openapi_object, "requestBody", Formatter.request_body(request))
end
```

File: lib/swagger/formatter.ex (hunk 1 above)
Comment (Danielwsx64) [reply lib/swagger/formatter.ex]: At next line there is a `Enum.reduce` with the same var name `acc`. =/

File: lib/swagger/formatter.ex (hunk 2 above)
Comment (Danielwsx64) [reply lib/swagger/formatter.ex]: Done =D

File: lib/swagger/formatter.ex [hunk 7]
```diff
@@ -2,45 +2,26 @@ defmodule Xcribe.Swagger.Formatter do
...
-            ),
-          "description" => Descriptor.get_param_description(name, controller, action),
-          "required" => Map.has_key?(params, name),
-          "schema" => %{"type" => type_of(value)}
-        }
-        | acc
-      ]
-    end)
+  import Xcribe.Swagger.Descriptor,
+    only: [get_param_description: 3, get_content_type: 1, get_attr_description: 2]
```
[the remark this answers is not in the document]
Comment (Danielwsx64) [reply lib/swagger/formatter.ex]: Nice =D

File: lib/swagger/swagger.ex [hunk 8]
```diff
@@ -2,127 +2,122 @@ defmodule Xcribe.Swagger do
...
+      "api_key" => %{"name" => "Authorization", "type" => "apiKey", "in" => "header"}
+    }
+  }
+
   def generate_doc(requests) do
-    swagger_json()
-    |> add_requests(requests)
-    |> add_security(requests)
+    requests
+    |> Enum.sort_by(& &1.status_code)
```
[the remark this answers is not in the document]
Comment (Danielwsx64) [reply lib/swagger/swagger.ex]: It looks much better.. Done  =D

File: lib/swagger/swagger.ex (hunk 3 above)
Comment (Danielwsx64) [reply lib/swagger/swagger.ex]: I usually prefer pattern matching as you suggested. But in this case the functions behaviour is very different. So I tried give a suggestive name for each. Maybe the current names are not so good. wdyt?

File: lib/swagger/swagger.ex (hunk 4 above)
Comment (Danielwsx64) [reply lib/swagger/swagger.ex]: Done :+1:

File: lib/swagger/swagger.ex (hunk 5 above)
Comment (Danielwsx64) [reply lib/swagger/swagger.ex]: :+1:

File: lib/swagger/swagger.ex (hunk 6 above)
Comment (Danielwsx64) [reply lib/swagger/swagger.ex]: done  =D

File: lib/swagger/swagger.ex (hunk 3 above)
Comment (iagoEffting) [reply lib/swagger/swagger.ex]: Got it. 

So the two functions are different proposes?
If yes, I agree to set a different name.

My problem is with `into`, that word breaks the reading.

WDYT 

- include_request_by_paths
- include_request_by_given_path

It's almost what you put, but, maybe the reading sounds better.

If you prefer `into` it's ok for me too. =D

Comment (Danielwsx64) [reply lib/swagger/swagger.ex]: You have a point. The names was bad. I improved the as:

include_request_into_paths           ->  build_path_from_request
include_request_into_given_path  -> build_path_verb_from_request
include_request_into_given_verb  -> build_responses_from_request

**Pull request merged**